pmd-core/src/main/java/net/sourceforge/pmd/PmdAnalysis.java

```diff
@@ -389,7 +389,7 @@ void performAnalysisImpl(List<? extends GlobalReportBuilderListener> extraListen
             } catch (Exception e) {
                 reporter.errorEx("Exception while closing analysis listeners", e);
                 // todo better exception
-                throw new RuntimeException("Exception while closing analysis listeners", e);
```

Comment:
I disagree with the code change, since all the other major exceptions have already been caught and stored in the `reporter`, it is totally fine to throw a `RuntimeException` in the final block, which will not hide the other exceptions. Without this exception, there will be duplicated exceptions thrown out as what has already been stored in the reporter. This will make the build log hard to track.